src/colmap/exe/sfm.cc

```diff
@@ -339,6 +357,143 @@ int RunHierarchicalMapper(int argc, char** argv) {
   return EXIT_SUCCESS;
 }
 
+int RunPosePriorMapper(int argc, char** argv) {
+  std::string input_path;
+  std::string output_path;
+
+  bool set_database_priors_covariance = false;
+  double prior_position_std_x = 1.;
+  double prior_position_std_y = 1.;
+  double prior_position_std_z = 1.;
+
+  OptionManager options;
+  options.AddDatabaseOptions();
+  options.AddImageOptions();
+  options.AddDefaultOption("input_path", &input_path);
+  options.AddRequiredOption("output_path", &output_path);
+  options.AddMapperOptions();
+
+  options.mapper->use_prior_position = true;
+
+  options.AddDefaultOption("set_database_priors_covariance",
```

Comment:
Is the following maybe a more intuitive option name? If I understand correctly, if this is false, then the covariance is read from the database?
```suggestion
  options.AddDefaultOption("overwrite_priors_covariance",
```

Reply:
Sounds good to me :+1: 

Yes for now, the covariance is always read from the database but enabling the option overwrites the current state of the priors covariance.
I am planning to later add more ways for setting up the priors covariance (python / c++ scripts to set it from a file and maybe also a button in the database_widget).